presto-parquet/src/main/java/com/facebook/presto/parquet/reader/PageReader.java

```diff
@@ -30,21 +29,23 @@
 {
     private final CompressionCodecName codec;
     private final long valueCount;
-    private final List<DataPage> compressedPages;
+    private final LinkedList<DataPage> compressedPages;
```

Comment:
what's the motivation of using LinkedList here? am inclined to use List interface, instead of specific implementations for compressedPages

Reply:
The implementation was already specifically constructing a LinkedList for this field, it was just doing it by creating one off of the input list inside of the constructor. The rationale was probably to make sure that the queue-like removal from the front pattern that it used was fast, cheap, and reclaimed memory as it proceeded. Anyway, this change preserves that behavior, but also shows allows calling a more efficient removeFirst() implementation when consuming and doesn’t require duplicating the list creation (since it was only ever constructed from a single place anyway).